tests/e2e/test_pipeline.py

```
@@ -19,7 +21,13 @@ def test_multiple_pipeline(shell_session):
 
 def test_pipeline_many_commands(shell_session):
     pipe_command = "echo Hello"
-    for _ in range(1000):
+
+    if os.uname().sysname == "Darwin":
+        num_pipes = 300
+    else:
+        num_pipes = 1000
```

Ruff

Use ternary operator `num_pipes = 300 if os.uname().sysname == "Darwin" else 1000` instead of `if`-`else`-block

Replace `if`-`else`-block with `num_pipes = 300 if os.uname().sysname == "Darwin" else 1000`

(SIM108)